### PR TITLE
#165955694 Build out PATCH /todos/:todoId endpoint

### PR DESCRIPTION
**Pull request by frankly034**
#### What does this PR do?
Have PATCH /todos/:todoId endpoint

#### Description of Task to be completed?
- add modify() to Todo model class 
- create PATCH /todos/:todoId route
- add verifyTokenMiddleware to PATCH /todos/:todoId and POST /todos routes
- add createTodo() to Validator class
- add createTodo middleware to POST /todos route

#### How should this be manually tested?
- Clone this branch
- Run ```npm run start-dev```
- Send a PATCH request to ```http://127.0.0.1:3000/api/v1/todos/:todoId```

#### Any background context you want to provide?
- You need node and npm installed
- Run ```npm install``` to install all dependencies 
- You need Postgres database
- Run ```cd src/db && sequelize db:migrate```
- Post data model ```
{
	"title": string,
	"description": string",
	"completed": boolean 
}```

#### What are the relevant pivotal tracker stories?
- #165955694 (https://www.pivotaltracker.com/story/show/165955694)

### CODE REVIEW

**Comment by mosinmiloluwa01** [comment]
hi Lewis,

Nice work so far, well done.

But i think you can check if a todo exists and then update it using todo.update instead of doing find by pk.

Also i think you are to verify that it is that user that owns the todo before u proceed to update the todo.

**Comment by frankly034** [comment]
Thanks @mosinmiloluwa01. 

I'll get right to it.

**Review by EmmanuelSage — COMMENTED** [review]
Nice one Lewis,
I can see the changes you made,
would you remove the completed rule from the validation as you are not using it again?

Asides the above question, it looks good to me

**Review by Sojisoyoye — COMMENTED** [review]
@frankly034, good job...looks cool

File: src/server/controllers/todo.js
Comment (Eazybee) [inline src/server/controllers/todo.js]: i think we might need to test this point together because i have some  concern i can't really explain until testes

File: src/server/controllers/todo.js
Comment (Eazybee) [inline src/server/controllers/todo.js]: as well as this point

File: src/server/middlewares/validator.js
Comment (Eazybee) [inline src/server/middlewares/validator.js]: Optional Tips: you might want to use object destructuring here

**Comment by akinyeleolat** [comment]
Once the merge conflict is sorted out, i think LGTM

File: src/server/middlewares/validator.js
Comment (Eazybee) [inline src/server/middlewares/validator.js]: I don't think ```max:5``` should be here. 
completed is supposed to be boolean.

You might want to check validatorjs documentation for checking boolean.
Another opinion is all new todo should be false by default when created

File: src/server/middlewares/validator.js
Comment (Eazybee) [inline src/server/middlewares/validator.js:77]: I reviewed ```todoController.create()``` in  ```controllers/todo.js``` and found out that in your controller, you used ```const { userId } = req.params;```  but you didn't validate the param your ```createTodo()``` in ```validator.js```

**Review by Eazybee — CHANGES_REQUESTED** [review]
Am really impressed by your pull request and I understand the logic behind it. However, there are some things I think you need to look into to give your code more quality

File: src/server/middlewares/validator.js
Comment (frankly034) [reply src/server/middlewares/validator.js:77]: I've made the changes

**Comment by Sojisoyoye** [comment]
@frankly034, Good job so far, especially by implementing all the feedbacks @Eazybee gave. Who is resolving the merge conflicts?

**Review by Eazybee — APPROVED** [review]

**Comment by Eazybee** [comment]
Please resolve conflict before merging

**Pull request merged**
